### ISSUE TITLE
MD018: false positive on paragraph continuation lines starting with #

### ISSUE THREAD

**Issue by ractive** [joshrotenberg/mdbook-lint#491]
> [!NOTE]
> This issue was generated by **Claude Code** (an AI agent) on behalf of [@ractive](https://github.com/ractive), hyalo's maintainer, who reviewed and authorized filing it. The reproduction was re-verified against `mdbook-lint-core` 0.15.2 immediately before posting.

**Version:** 0.15.2 (also present in 0.14).

MD018 ("No space after hash on atx style heading") fires on a wrapped paragraph whose *continuation* line happens to begin with `#` — for example a bare issue reference such as `#472`. That line is paragraph text, not a heading, and `#472` is not a valid ATX heading under CommonMark anyway (ATX requires a space after the `#` sequence).

## Reproduction

```markdown
Upstream fixed this in PR
#472 and shipped it in 0.15.2.
```

MD018 reports a violation on line 2. Expected: no violation — line 2 is a lazy continuation line of the paragraph started on line 1, so CommonMark never parses it as a heading.

Contrast:

- `#foo` alone between blank lines — correctly flagged (it *is* a malformed heading).
- `PR #472` mid-line — correctly not flagged.
- `#472` as a continuation line — **falsely flagged**.

## Notes

Verified still present in 0.15.2 by diffing `md018.rs` against 0.14: only the fix generation and the column unit changed; the detection logic did not. The rule appears to scan lines rather than consult block structure, so it cannot see that the line is inside a paragraph.

Precedent for the shape of this bug: #274 ("MD018: false positive on Rust attributes inside code blocks") was accepted and fixed.

## Why this matters for embedders

[hyalo](https://github.com/ractive/hyalo) embeds `mdbook-lint-core` as a library and runs it over documentation vaults where bare issue references at line starts are common (wrapped changelog/plan prose). The workaround available downstream is to disable MD018 wholesale, which also loses the genuine `#Heading` typo detection the rule exists for.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)
